src/otx/core/data/entity/base.py

```diff
@@ -0,0 +1,157 @@
+        2) `image_type=ImageType.TV_IMAGE`: C x H x W torchvision image tensor
+    :param imgs_info: Meta information for images
+    """
+
+    image: np.ndarray | tv_tensors.Image
+    img_info: ImageInfo
+
+    @property
+    def task(self) -> OTXTaskType:
+        """OTX task type definition."""
+        msg = "OTXTaskType is not defined."
+        raise RuntimeError(msg)
+
+    @property
+    def image_type(self) -> ImageType:
+        """Image type definition."""
+        if isinstance(self.image, np.ndarray):
+            return ImageType.NUMPY
+        if isinstance(self.image, tv_tensors.Image):
+            return ImageType.TV_IMAGE
+
+        raise TypeError(self.image)
+
+
+@dataclass
+class OTXPredEntity(OTXDataEntity):
+    """Data entity to represent the model output prediction."""
+    score: np.ndarray | Tensor
+
+
+T_OTXBatchDataEntity = TypeVar(
+    "T_OTXBatchDataEntity",
+    bound="OTXBatchDataEntity",
+)
+
+
+@dataclass
+class OTXBatchDataEntity(Generic[T_OTXDataEntity]):
+    """Base Batch data entity for OTX.
+
+    This entity is the output of PyTorch DataLoader,
+    which is the direct input of OTXModel.
+
+    :param images: List of B numpy image tensors (C x H x W)
+    :param imgs_info: Meta information for images
+    """
+
+    batch_size: int
+    images: list[tv_tensors.Image]
+    imgs_info: list[ImageInfo]
+
+    @property
+    def task(self) -> OTXTaskType:
+        """OTX task type definition."""
+        msg = "OTXTaskType is not defined."
+        raise RuntimeError(msg)
+
+    @classmethod
+    def collate_fn(cls, entities: list[T_OTXDataEntity]) -> OTXBatchDataEntity:
+        """Collection function to collect `OTXDataEntity` into `OTXBatchDataEntity` in data loader."""
+        if (batch_size := len(entities)) == 0:
+            msg = "collate_fn() input should have > 0 entities"
+            raise RuntimeError(msg)
+
+        task = entities[0].task
+
+        if not all(task == entity.task for entity in entities):
+            msg = "collate_fn() input should include a single OTX task"
+            raise RuntimeError(msg)
+
+        if not all(entity.image_type == ImageType.TV_IMAGE for entity in entities):
+            msg = "All entities should be torchvision's Image tensor before collate_fn()"
+            raise RuntimeError(msg)
+
+        return OTXBatchDataEntity(
+            batch_size=batch_size,
+            images=[entity.image for entity in entities],
+            imgs_info=[entity.img_info for entity in entities],
+        )
+
+
+T_OTXBatchPredEntity = TypeVar(
+    "T_OTXBatchPredEntity",
+    bound="OTXBatchPredEntity",
+)
+
+
+@dataclass
+class OTXBatchPredEntity(OTXBatchDataEntity):
+    """Data entity to represent model output predictions."""
+    scores: list[np.ndarray] | list[Tensor]
+
+
+T_OTXBatchLossEntity = TypeVar(
+    "T_OTXBatchLossEntity",
+    bound="OTXBatchLossEntity",
+)
+
+
+class OTXBatchLossEntity(Dict[str, Tensor]):
```

Comment:
What is the rationale behind this specific datatype? Is it simply a type alias for `Dict[str, Tensor]`? If it doesn't offer any additional functionality, wouldn't it be more straightforward to use `Dict[str, Tensor]` directly?

Reply:
I leaved it as is for the future extension.